addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4908,6 +4908,8 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
             this.$weight.text(`${this._filesize.toFixed(1)} kb`);
             this.$weight.removeClass('d-none');
             this._relocateWeightEl();
+        } else {
+            this.$weight.addClass('d-none');
```

Comment:
You can add this in the first check on `this._filesize` instead of `else`:

```javascript
if (this._filesize === undefined) {
    this.$weight.addClass('d-none');
    await this._applyOptions(false);
}
if (this._filesize !== undefined) {
    this.$weight.text(`${this._filesize.toFixed(1)} kb`);
    this.$weight.removeClass('d-none');
    this._relocateWeightEl();
}
```